flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/StreamSourceContexts.java

```diff
@@ -149,15 +142,16 @@ public void close() {}
 
         private long lastRecordTime;
 
+        private volatile boolean idle = false;
+
         private AutomaticWatermarkContext(
                 final Output<StreamRecord<T>> output,
                 final long watermarkInterval,
                 final ProcessingTimeService timeService,
                 final Object checkpointLock,
-                final StreamStatusMaintainer streamStatusMaintainer,
                 final long idleTimeout) {
 
-            super(timeService, checkpointLock, streamStatusMaintainer, idleTimeout);
+            super(timeService, checkpointLock, idleTimeout);
```

Comment:
Removing stream status maintainer feels like an independent change. Can we move that into a separate commit? It makes reviewing harder.